luda-editor/luda-editor-client/src/app/editor/clip_editor/camera_clip_editor/image_browser/mod.rs

```diff
@@ -4,22 +4,34 @@ mod browser_item;
 use browser_item::*;
 mod back_button;
 mod scroll;
-use crate::app::{editor::events::EditorEvent, types::ImageFilenameObject};
+use crate::app::{
+    editor::events::EditorEvent,
+    types::{CharacterPoseEmotion, ImageFilenameObject},
+};
 use scroll::*;
+mod types;
+pub use types::*;
 
 #[derive(Debug)]
 pub struct ImageBrowser {
-    directory_key: String,
-    selected_key: Option<String>,
+    directory: ImageBrowserDirectory,
+    selected_item: Option<ImageBrowserItem>,
     current_directory_label_layout: XywhRect<f32>,
     scroll: Scroll,
 }
 
 impl ImageBrowser {
-    pub fn new() -> Self {
+    pub fn new(character_pose_emotion: &CharacterPoseEmotion) -> Self {
         Self {
-            directory_key: "".to_string(),
-            selected_key: None,
+            directory: ImageBrowserDirectory::CharacterPose(
```

Comment:
This is the code when user open the image browser. I set first opened directory as image's directory.